tmva/sofie/inc/TMVA/ROperator_Max.hxx

```diff
@@ -0,0 +1,101 @@
+#ifndef TMVA_SOFIE_ROPERATOR_MAX
+#define TMVA_SOFIE_ROPERATOR_MAX
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <sstream>
+#include <algorithm>
+
+namespace TMVA{
+namespace Experimental{
+namespace SOFIE{
+
+template <typename T>
+class ROperator_Max final : public ROperator
+{
+
+private:
+
+   std::vector<std::string> fInputNames;
+   std::string fNY;
+   std::vector<std::vector<size_t> > fShapeInputs;
+   std::vector<size_t> fShape;
+
+public:
+   ROperator_Max(){}
+   ROperator_Max( const std::vector<std::string> & inputNames, std::string nameY):
+   fNY(UTILITY::Clean_name(nameY)){
+      fInputNames.reserve(inputNames.size());
+      for (auto & name : inputNames)
+         fInputNames.push_back(UTILITY::Clean_name(name));
+   }
+
+   // type of output given input 
+   std::vector<ETensorType> TypeInference(std::vector<ETensorType> input){
+      return input;
+   }
+
+   // shape of output tensors given input tensors
+   std::vector<std::vector<size_t>> ShapeInference(std::vector<std::vector<size_t>> input){
+      auto ret = std::vector<std::vector<size_t>>(1, input[0]); 
+      return ret;
+   }
+
+   void Initialize(RModel& model){
+      // input must be a graph input, or already initialized intermediate tensor
+
+      for (auto &it : fInputNames) {
+         if (model.CheckIfTensorAlreadyExist(it) == false) {
+            throw std::runtime_error("TMVA SOFIE Concat Op Input Tensor " + it + " is not found in model");
+         }
+         fShapeInputs.push_back(model.GetTensorShape(it));
+      }
+      for(size_t i=0; i < fShapeInputs.size()-1; i++){
+         if(fShapeInputs[i] != fShapeInputs[i+1]){
+            std::string msg = "TMVA SOFIE Max Op: Support only inputs with same shape";
```

Comment:
Here we support only inputs to Max with the same shape. 
we should support Broadcasting, at least to tensor with same length as we are doing for the Binary operators

Reply:
Done with the changes and tested locally